src/main/java/com/hotels/bdp/circustrain/bigquery/partition/BigQueryPartitionGenerator.java

```diff
@@ -97,6 +97,7 @@ private ExtractionUri scheduleForExtraction(com.google.cloud.bigquery.Table tabl
     ExtractionUri extractionUri = new ExtractionUri(destinationBucket, generateFolderName(), generateFileName());
     ExtractionContainer toRegister = new ExtractionContainer(table, extractionUri, PostExtractionAction.DELETE);
     extractionService.register(toRegister);
+    extractionService.extract();
```

Comment:
The method name is a bit misleading now as it's not just scheduling for extraction, it's actually doing the extraction. Feels a bit weird that we're now doing this extraction in two places.